compiler/parampatterns.nim

```diff
@@ -254,7 +258,7 @@ proc isAssignable*(owner: PSym, n: PNode; isUnsafeAddr=false): TAssignableResult
       # types that are equal modulo distinction preserve l-value:
       result = isAssignable(owner, n.sons[1], isUnsafeAddr)
   of nkHiddenDeref:
-    if n[0].typ.kind == tyLent: result = arDiscriminant
```

Comment:
You need to watch out that ``returnsLent().field = value`` and ``passToVarT(returnsLent())`` is not allowed. ``arDiscriminant`` was supposed to cover that.

Reply:
Good catch, missed the fact that else is lValue. Fixed. Extended test to cover this case